packages/gitbook/src/app/(site)/layout.tsx

```diff
@@ -1,18 +1,30 @@
 import { CustomizationRootLayout } from '@/components/RootLayout';
 import { getSiteData } from '@/lib/api';
+import { checkIsFromMiddleware } from '@/lib/pages';
 import { getSiteContentPointer } from '@/lib/pointer';
 
 /**
  * Layout to be used for the site root. It fetches the customization data for the site
  * and initializes the CustomizationRootLayout with it.
  */
 export default async function SiteRootLayout(props: { children: React.ReactNode }) {
-    const { children } = props;
+    const fromMiddleware = await checkIsFromMiddleware();
+    if (!fromMiddleware) {
+        return (
+            <html lang="en">
+                <body className="font-[sans-serif]">
+                    <main>{props.children}</main>
+                </body>
+            </html>
+        );
+    }
```

Comment:
While it works, I think having to handle it in every single route/layout that we have is expensive.

Another solution could have been:

- When the middleware does the `NextResponse.rewrite`, we put a prefix to the path like `middleware`
- We wrap all the routes in a folder `src/app/middleware`
- We put a not found / layout outside for the case that are not handled

Reply:
Done, the default 404 for Vercel is good enough.